UnBanned user event

## What did you do?
The user was unbanned from all channels using server-side

## What did you expect to happen?
Get event userUnbanned

## What happened instead?
Decoding Error

## GetStream Environment
**GetStream Chat version:** 2.2.4
**GetStream Chat frameworks:** StreamChatClient, StreamChatCore, StreamChat
**iOS version:** 13.4
**Swift version:** Swift 5
**Xcode version:** 11.4.1
**Device:** iPhone 11 Pro Simulator

## Additional context
StreamChatClient WebSocket.swift
Decode response dataCorrupted(Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "type", intValue: nil)], debugDescription: "Cannot initialize EventType from invalid String value user.unbanned", underlyingError: nil)) in parseEvent(with:)[366]
[16 Jun 13:43:29.640] [ERROR] 📦  {
  "channel_id" : "1285",
  "channel_type" : "messaging",
  "type" : "user.unbanned",
  "created_at" : "2020-06-16T06:43:29.554149291Z"
}

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### ✅ Added
 - `Message.translate` for message translations. Please see [docs](https://getstream.io/chat/docs/translation/?language=swift) for more info [#319](https://github.com/GetStream/stream-chat-swift/issues/319)
+- `user.unbanned` event added [#319](https://github.com/GetStream/stream-chat-swift/issues/319)
 
 # [2.2.4](https://github.com/GetStream/stream-chat-swift/releases/tag/2.2.4)
 _June 12, 2020_

--- a/Sources/Client/Client/Client+WebSocket.swift
+++ b/Sources/Client/Client/Client+WebSocket.swift
@@ -94,6 +94,19 @@ extension Client: WebSocketEventDelegate {
             userAtomic.set(user)
             return true
             
+        case .userUnbanned(let user, let cid, _, _):
+            if let cid = cid {
+                watchingChannelsAtomic.update { (channelsByCid) -> [ChannelId: [WeakRef<Channel>]] in
+                    var channelsByCid = channelsByCid
+                    if let weakChannels = channelsByCid[cid] {
+                        weakChannels.forEach({ $0.value?.bannedUsers.removeAll(where: { $0.id ==  user.id }) })
+                        channelsByCid[cid] = weakChannels
+                    }
+                    return channelsByCid
+                }
+            }
+            return true
+            
         case let .messageNew(message, _, _, _) where message.user != user && user.isMuted(user: message.user):
             // FIXIT: This shouldn't be by default.
             logger?.log("Skip a message (\(message.id)) from muted user (\(message.user.id)): \(message.textOrArgs)", level: .info)

--- a/Sources/Client/Model/Event.swift
+++ b/Sources/Client/Model/Event.swift
@@ -46,6 +46,8 @@ public enum Event: Decodable, Equatable {
     case userUpdated(User, ChannelId?, EventType)
     /// When a user was banned (when subscribed to the user presence).
     case userBanned(reason: String?, expiration: Date?, created: Date, ChannelId?, EventType)
+    /// When a user was unbanned (when subscribed to the user presence).
+    case userUnbanned(User, ChannelId?, created: Date, EventType)
     /// When a user starts watching a channel (when watching the channel).
     case userStartWatching(User, _ watcherCount: Int, ChannelId?, EventType)
     /// When a user stops watching a channel (when watching the channel).
@@ -132,6 +134,7 @@ public enum Event: Decodable, Equatable {
              .userStartWatching(_, _, _, let type),
              .userStopWatching(_, _, _, let type),
              .userBanned(_, _, _, _, let type),
+             .userUnbanned(_, _, _, let type),
              
              .memberAdded(_, _, let type),
              .memberUpdated(_, _, let type),
@@ -181,6 +184,7 @@ public enum Event: Decodable, Equatable {
              .userStartWatching(_, _, let cid, _),
              .userStopWatching(_, _, let cid, _),
              .userBanned(_, _, _, let cid, _),
+             .userUnbanned(_, let cid, _, _),
              
              .memberAdded(_, let cid, _),
              .memberUpdated(_, let cid, _),
@@ -216,6 +220,7 @@ public enum Event: Decodable, Equatable {
              .userPresenceChanged(let user, _, _),
              .userStartWatching(let user, _, _, _),
              .userStopWatching(let user, _, _, _),
+             .userUnbanned(let user, _, _, _),
              .memberRemoved(let user, _, _),
              .reactionNew(_, _, let user, _, _),
              .reactionUpdated(_, _, let user, _, _),
@@ -358,6 +363,15 @@ public enum Event: Decodable, Equatable {
             let reason = try container.decodeIfPresent(String.self, forKey: .reason)
             let expiration = try container.decodeIfPresent(Date.self, forKey: .expiration)
             self = try .userBanned(reason: reason, expiration: expiration, created: created(), channelId, type)
+        case .userUnbanned:
+            var channelId: ChannelId? = try? cid()
+            
+            if let channelType = try container.decodeIfPresent(ChannelType.self, forKey: .channelType),
+                let id = try container.decodeIfPresent(String.self, forKey: .channelId) {
+                channelId = ChannelId(type: channelType, id: id)
+            }
+            
+            self = try .userUnbanned(user(), channelId, created: created(), type)
             
         // Member
         case .memberAdded:

--- a/Sources/Client/Model/EventType.swift
+++ b/Sources/Client/Model/EventType.swift
@@ -24,6 +24,8 @@ public enum EventType: String, Codable, CaseIterable {
     case userUpdated = "user.updated"
     /// When a user was banned (when subscribed to the user presence).
     case userBanned = "user.banned"
+    /// When a user was unbanned (when subscribed to the user presence).
+    case userUnbanned = "user.unbanned"
     /// When a user starts watching a channel (when watching the channel).
     case userStartWatching = "user.watching.start"
     /// When a user stops watching a channel (when watching the channel).

--- a/Tests/Client/Unit Tests/EventTests.swift
+++ b/Tests/Client/Unit Tests/EventTests.swift
@@ -75,6 +75,7 @@ final class EventTests: XCTestCase {
              .userPresenceChanged,
              .userUpdated,
              .userBanned,
+             .userUnbanned,
              .userStartWatching,
              .userStopWatching,
              .typingStart,

